packages/core/content-manager/server/controllers/relations.js

```diff
@@ -186,39 +186,43 @@ module.exports = {
     }
 
     const queryParams = {
-      sort: mainField,
-      ...query,
-      fields: fieldsToSelect, // cannot select other fields as the user may not have the permissions
-      filters: {}, // cannot filter for RBAC reasons
+      select: fieldsToSelect,
     };
 
     if (!isEmpty(idsToOmit)) {
-      addFiltersClause(queryParams, { id: { $notIn: idsToOmit } });
-    }
-
-    // searching should be allowed only on mainField for permission reasons
-    if (_q) {
-      addFiltersClause(queryParams, { [mainField]: { $containsi: _q } });
+      queryParams.filters = { id: { $notIn: idsToOmit } };
```

Comment:
Could we add a comment here explaining why we have idsToOmit?
Which use case does it cover? When the frontend has deleted a relationship but not saved?